### PR TITLE
Pin all crate dependency versions

### LINKED ISSUE #362: Crate dependency versions aren't being correctly pinned
**Steps to reproduce**

1. Run `./library/crates/update.sh` without modifying `library/crates/Cargo.toml`

**Expected Output**

No-op

**Actual Output**

Some of the dependencies get bumped to later versions - even those that we have specified in our Cargo.toml, such as `cxx`! (We currently specify `cxx = "1.0.59"`, yet `cargo raze` bumps the actual installed version to "1.0.72".)

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

We've pinned the version numbers of all crate dependencies.

## What are the changes implemented in this PR?

Previously crate dependencies were free to increment their own patch versions - the default in `Cargo.toml`. Now we prefix each dependency version with the `=` modifier, pinning the precise version. This brings the behaviour of our crate dependencies in line with our Maven dependencies.

Closes #362 .

### CODE REVIEW

**Comment by grabl** [comment]
### PR Review Checklist

Do not edit the content of this comment. The PR reviewer should simply update this comment by ticking each review item below, as they get completed.

----
**Code**

- [ ] Packages, classes, and methods have a single domain of responsibility.
- [ ] Packages, classes, and methods are grouped into cohesive and consistent domain model.
- [ ] The code is canonical and the minimum required to achieve the goal.
- [ ] Modules, libraries, and APIs are easy to use, robust (foolproof and not errorprone), and tested.
- [ ] Logic and naming has clear narrative that communicates the accurate intent and responsibility of each module (e.g. method, class, etc.).
- [ ] The code is algorithmically efficient and scalable for the whole application.

**Architecture**

- [ ] Any required refactoring is completed, and the architecture does not introduce technical debt incidentally.
- [ ] Any required build and release automations are updated and/or implemented.
- [ ] Any new components follows a consistent style with respect to the pre-existing codebase.
- [ ] The architecture intuitively reflects the application domain, and is easy to understand.
- [ ] The architecture has a well-defined hierarchy of encapsulated components.
- [ ] The architecture is extensible and scalable.

File: library/crates/BUILD.bazel [hunk 1]
```diff
@@ -41,7 +41,7 @@ alias(
```
Comment (alexjpwalker) [inline library/crates/BUILD.bazel:41]: All changes are generated code, except for `Cargo.toml`.

File: library/crates/Cargo.toml [hunk 2]
```diff
@@ -6,26 +6,26 @@ version = "0.0.0"
...
-protobuf = { version = "2.18.2", features = ["with-bytes"] }
-protobuf-codegen = "2.18.2"
-protoc-rust-grpc = "0.8.3"
-rocksdb = "0.17.0"
-tokio = { version = "1.20.1", features = ["full"] }
-uuid = { version = "1.1.2", features = ["fast-rng", "v4"] }
-tonic = "0.8.0"
-prost = "0.11.0"
-tonic-build = "0.8.0"
+antlr-rust = "=0.3.0-beta"
```
Comment (alexjpwalker) [inline library/crates/Cargo.toml:9]: `=` modifier pins the precise version.

File: library/crates/Cargo.toml [hunk 3]
```diff
@@ -6,26 +6,26 @@ version = "0.0.0"
...
+axum = "=0.5.15"
+chrono = "=0.4.22"
+cxx = "=1.0.59"
+derivative = "=2.2.0"
+enum_dispatch = "=0.3.8"
+futures = { version = "=0.3.21", features = ["executor", "thread-pool"] }
+grpc = "=0.8.3"
+grpc-compiler = "=0.8.3"
+grpc-protobuf = "=0.8.3"
+log = "=0.4.8"
```
Comment (alexjpwalker) [inline library/crates/Cargo.toml:19]: We had to bump `log` to `0.4.8` because another crate required at least this version.

**Review by alexjpwalker — COMMENTED** [review]
These changes look good

**Review by dmitrii-ubskii — APPROVED** [review]
Seems straightforward.

**Pull request merged**